src/Select.js

```diff
@@ -1066,6 +1072,51 @@ class Select extends React.Component {
 		);
 	}
 
+	renderUniqueSelect(className, valueArray) {
```

Comment:
Please try to avoid code duplication. This function repeats a lot of code from what is already in `render`.